utils/funcexp/func_round.cpp

```diff
@@ -136,18 +138,7 @@ int64_t Func_round::getIntVal(Row& row, FunctionParm& parm, bool& isNull,
 uint64_t Func_round::getUintVal(Row& row, FunctionParm& parm, bool& isNull,
                                 CalpontSystemCatalog::ColType& op_ct)
 {
-  uint64_t x;
-  if (UNLIKELY(op_ct.colDataType == execplan::CalpontSystemCatalog::DATE))
-  {
-    IDB_Decimal d = getDecimalVal(row, parm, isNull, op_ct);
-    x = static_cast<uint64_t>(d.value);
-  }
-  else
-  {
-    x = parm[0]->data()->getUintVal(row, isNull);
-  }
-
-  return x;
+  return static_cast<uint64_t>(getIntVal(row, parm, isNull, op_ct));
```

Comment:
Can you ellaborate on this?

Reply:
Yep, verified with the server - we have to return 0 for CAST(-1 AS UNSIGNED INTEGER) if -1 is wide decimal.